tests/Jobs/PlatformStatsSummaryJobTest.php

```diff
@@ -16,28 +16,29 @@
 
 class PlatformStatsSummaryJobTest extends TestCase
 {
-    use DatabaseTransactions;
+    use RefreshDatabase;
 
     private $numWikis = 5;
     private $wikis = [];
+    private $users = [];
 
     private $db_prefix = "somecoolprefix";
     private $db_name = "some_cool_db_name";
 
     protected function setUp(): void {
         parent::setUp();
-        for($n = 0; $n < $this->numWikis; $n++ ) {
+        for ($n = 0; $n < $this->numWikis; $n++) {
             DB::connection('mysql')->getPdo()->exec("DROP DATABASE IF EXISTS {$this->db_name}{$n};");
         }
-        $this->seedWikis();
-        $this->manager = $this->app->make('db');
+        $this->wikis = [];
+        $this->users = [];
     }
 
     protected function tearDown(): void {
-        foreach($this->wikis as $wiki) {
-            $wiki['wiki']->wikiDb()->forceDelete();
-            $wiki['wiki']->forceDelete();
-        }
+        Wiki::query()->delete();
+        User::query()->delete();
+        WikiManager::query()->delete();
+        WikiDb::query()->delete();
```

Comment:
This sounds familiar - I think that even may be the reason why we have this target in the Makefile to run tests with a clean database https://github.com/wbstack/api/blob/main/Makefile#L11

(I'm not trying to say this is a good thing - something doesn't seem to work there)

Reply:
I found another trait called `DatabaseMigrations`: https://laravel.com/docs/8.x/dusk#migrations

It seems we don't need the extended `tearDown` method when using this trait. (at least the test completes successfully repeatedly with it for me)

I'm not 100% certain this is the right thing to do, as I found this in a browser automation testing section in the laravel docs.

Reply:
Not sure what Dusk even is? It might be worth spending some time in removing all statefulness from all tests at some point. It's kind of suboptimal this bites me every single time I add a feature and spend a lot of time debugging why my new test now breaks another test. Ideally, we could even aim for having the tests run using an in-memory database so that we can just run them locally without a database proper.

Reply:
There has been work on this here https://github.com/wbstack/api/pull/559 but I'm not sure why it was abandoned.